src/software/ai/hl/stp/play/play.cpp

```diff
@@ -28,17 +20,42 @@ PriorityTacticVector Play::getTactics(const World &world)
     {
         // Run the coroutine. This will call the bound getNextTactics function
         tactic_sequence();
+    }
+    else
+    {
+        tactic_sequence = TacticCoroutine::pull_type(
+            boost::bind(&Play::getNextTacticsWrapper, this, _1));
+        // Run the coroutine. This will call the bound getNextTactics function
+        tactic_sequence();
+    }
 
-        // Check if the coroutine is still valid before getting the result. This makes
-        // sure we don't try get the result after "running out the bottom" of the
-        // coroutine function
+    // Check if the coroutine is still valid before getting the result. This makes
+    // sure we don't try get the result after "running out the bottom" of the
+    // coroutine function
+    if (tactic_sequence)
+    {
+        // Extract the result from the coroutine. This will be whatever value was
+        // yielded by the getNextTactics function
+        auto next_tactics = tactic_sequence.get();
+        return next_tactics;
+    }
+    else
+    {
+        tactic_sequence = TacticCoroutine::pull_type(
+            boost::bind(&Play::getNextTacticsWrapper, this, _1));
+        // Run the coroutine. This will call the bound getNextTactics function
+        tactic_sequence();
```

Comment:
what's this bit doing here? a comment would be helpful

Reply:
added a comment, note that this is all going to go away eventually because we'll be getting rid of coroutines